SendLog.py

```diff
@@ -113,9 +190,26 @@ def event_stream(xcos_file_id):
 	# out will contain output of command, the list of process IDs of scilab
 	(out, err) = proc.communicate()
 	_l = len(out)
-	# Run xcos file
-	command = ["./"+SCI+"bin/scilab-adv-cli", "-nogui", "-noatomsautoload", "-nb", "-nw", "-e", "loadXcosLibs();importXcosDiagram('" + xcos_file_dir + xcos_file_name + "');xcos_simulate(scs_m,4);mode(2);quit()"]
-	scilab_proc = subprocess.Popen(command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, shell=False);
+	#initialise pid
+	pid = 0
+        #id to identify each session for saving workspace 
+        session=Details.uid
+        #name of worspace file the session
+        workspace="workspace"+session+".dat"
+        workspace_counter=workspace_list[xcos_file_id]
+        #commands for ruuning of scilab based on existence of TOWS_c and FROMWSB
+        #3 means both exists,2 FROMWSB exists,1 TOWS_c exists,0 none exists meaning normal set of commands 
+        if (workspace_counter ==3 and exists(workspace)):
+            append=workspace_dict[xcos_file_id]
+            command = ["./"+SCI+"bin/scilab-adv-cli", "-nogui", "-noatomsautoload", "-nb", "-nw", "-e","load('"+workspace+"');loadXcosLibs();importXcosDiagram('" + xcos_file_dir + xcos_file_name + "');xcos_simulate(scs_m,4);xs2jpg(gcf(),'webapp/res_imgs/img_test.jpg'),mode(2);deletefile('"+workspace+"');save('"+workspace+"') ;quit()"]
```

Comment:
- Avoid relative paths such as '..'  or '. /' etc
- command list is too long, please use only 80 columns width